Sources/GRPCCore/RPCDuration.swift

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2023, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// A representation of a period of time with nanosecond precision.
+///
+/// The largest representable duration is `Int64.max` nanoseconds.
+public struct RPCDuration: Sendable, Hashable {
```

Comment:
Why are we introducing our own duration here instead of using `Swift.Duration`? Even if we only care about nanoseconds precision we can still clamp the `Swift.Duration`. Just from user perspective I nowadays always assume to be able to put `Duration` into the APIs directly and the API will do the necessary clamping under the hood.